### PR TITLE
check  while ~/.kube/config is missing

### LINKED ISSUE #448: vcctl only reads the ~/.kube/config and neglects $KUBECONFIG
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind feature

**What happened**:
When I created a local k8s cluster using `~/kubernetes/hack/local-up-cluster.sh`, the guideline suggested me to `export KUBECONFIG=/var/run/kubernetes/admin.kubeconfig` instead of creating a new config in `~/.kube/`. Then the `vcctl` command line reported `Failed to list job: stat /root/.kube/config: no such file or directory` and failed to execute.
After I copyed the config file referred to $KUBECONFIG to `~/.kube/` , `vcctl` run correctly. So I doubt that the `vcctl` command line is set to read `~/.kube/config` only and neglects the enviroment variable $KUBECONFIG.

**What you expected to happen**:
When `vcctl` cannot find `~/.kube/config`, it shall get $KUBECONFIG as an alternative.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**: 
- Volcano Version: release-0.2
- Kubernetes version (use `kubectl version`): v1.15.3
- Cloud provider or hardware configuration: 
- OS (e.g. from /etc/os-release): Ubuntu 18.04
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by Rui-Tang**
fix #448

### CODE REVIEW

**Comment by k82cn** [comment]
/approve

Thanks for the fix :)

File: pkg/cli/job/common.go [hunk 1]
```diff
@@ -31,7 +31,14 @@ func initFlags(cmd *cobra.Command, cf *commonFlags) {
...
 	if home := homeDir(); home != "" {
-		cmd.Flags().StringVarP(&cf.Kubeconfig, "kubeconfig", "k", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+		/*
+			Default kubeconfig file is located in $HOME/.kube/config . In case this file is not exist, it will look for $KUBECONFIG instead.
+		*/
+		if defaultKubeConfigFile := filepath.Join(home, ".kube", "config"); checkFileExist(defaultKubeConfigFile) {
+			cmd.Flags().StringVarP(&cf.Kubeconfig, "kubeconfig", "k", defaultKubeConfigFile, "(optional) absolute path to the kubeconfig file")
+		} else if kubeConfig := kubeConfig(); kubeConfig != "" {
+			cmd.Flags().StringVarP(&cf.Kubeconfig, "kubeconfig", "k", kubeConfig, "(optional) absolute path to the kubeconfig file")
+		}
```
[the remark this answers is not in the document]
Comment (TommyLike) [reply pkg/cli/job/common.go]: It's possible that we end up missing the `Kubeconfig` option if the  if statement on line 39 is missed.

File: pkg/cli/job/common.go [hunk 2]
```diff
@@ -31,7 +31,14 @@ func initFlags(cmd *cobra.Command, cf *commonFlags) {
 	cmd.Flags().StringVarP(&cf.Master, "master", "s", "", "the address of apiserver")
 
 	if home := homeDir(); home != "" {
-		cmd.Flags().StringVarP(&cf.Kubeconfig, "kubeconfig", "k", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+		/*
+			Default kubeconfig file is located in $HOME/.kube/config . In case this file is not exist, it will look for $KUBECONFIG instead.
```
Comment (TommyLike) [inline pkg/cli/job/common.go]: does not exist

**Comment by TommyLike** [comment]
Please fix the Tests issues

File: pkg/cli/job/common.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rui-Tang) [reply pkg/cli/job/common.go]: I have reordered code and it should fix this problem.

File: pkg/cli/job/common.go [hunk 3]
```diff
@@ -31,7 +31,14 @@ func initFlags(cmd *cobra.Command, cf *commonFlags) {
...
+		*/
```
[the remark this answers is not in the document]
Comment (Rui-Tang) [reply pkg/cli/job/common.go]: I have changed it.

File: pkg/cli/job/common.go (hunk 2 above)
Comment (Rui-Tang) [reply pkg/cli/job/common.go]: thanks.

**Comment by Rui-Tang** [comment]
@TommyLike  I don't know why e2e test failed. Can you please look at it and tell me how to fix it?

File: pkg/cli/job/common.go [hunk 4]
```diff
@@ -30,9 +30,13 @@ type commonFlags struct {
...
 	cmd.Flags().StringVarP(&cf.Master, "master", "s", "", "the address of apiserver")
 
-	if home := homeDir(); home != "" {
-		cmd.Flags().StringVarP(&cf.Kubeconfig, "kubeconfig", "k", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	if defaultKubeConfigFile := filepath.Join(homeDir(), ".kube", "config"); checkFileExist(defaultKubeConfigFile) {
+		cmd.Flags().StringVarP(&cf.Kubeconfig, "kubeconfig", "k", defaultKubeConfigFile, "(optional) absolute path to the kubeconfig file")
+	} else if kubeConfig := kubeConfig(); kubeConfig != "" {
+		//Default kubeconfig file is located in $HOME/.kube/config . In case this file does not exist, it will look for $KUBECONFIG instead.
+		cmd.Flags().StringVarP(&cf.Kubeconfig, "kubeconfig", "k", kubeConfig, "(optional) absolute path to the kubeconfig file")
 	} else {
```
Comment (hzxuzhonghu) [inline pkg/cli/job/common.go]: @Rui-Tang The e2e failure is related to this. And the e2e in test/e2e/vcctl.go should be updated accordingly.
I would suggest the following orders

```
1. --kubeconfig=/path/to/.kube/config flag

2. KUBECONFIG  ENV

3. $HOME/.kube/config 
```

Comment (k82cn) [reply pkg/cli/job/common.go]: I merged this PR forcefully; @hzxuzhonghu , please help to fix e2e error :)

**Pull request merged**